src/main/java/net/dv8tion/jda/core/utils/PermissionUtil.java

```diff
@@ -420,79 +481,165 @@ public static long getEffectivePermission(Channel channel, Role role)
     {
-        //--Do we have all permissions possible? (Owner or member has ADMINISTRATOR permission)
-        //--If we have all permissions possible, then we will be able to see this room.
-        //WE DO NOT WANT TO CHECK THIS FOR CHANNELS, AS CHANNELS CAN OVERRIDE MANAGE_PERMISSIONS
-//        if (checkPermission(member, Permission.ADMINISTRATOR, guild))
-//            return true;
-
-        //BUT: WE DO WANT TO CHECK IF HE IS OWNER
-        if (guild.getOwner().equals(member))
-            return true;
+        Args.notNull(member, "Member");
+        Guild guild = member.getGuild();
+
+        long permission = guild.getPublicRole().getPermissionsRaw();
+
+        for (Role role : member.getRoles())
+            permission |= role.getPermissionsRaw();
 
-        long effectivePerms = getEffectivePermission(member, guild, roleOverrides, memberOverrides);
-        return ((effectivePerms & (1 << Permission.ADMINISTRATOR.getOffset())) | (effectivePerms & (1 << perm.getOffset()))) > 0;
+        return permission;
     }
 
-    private static long getEffectivePermission(Member member, GuildImpl guild, Map<Role, PermissionOverride> roleOverrides, Map<Member, PermissionOverride> memberOverrides)
+    /**
+     * Retrieves the implicit permissions of the specified {@link net.dv8tion.jda.core.entities.Member Member}
+     * in its hosting {@link net.dv8tion.jda.core.entities.Guild Guild} and specific {@link net.dv8tion.jda.core.entities.Channel Channel}.
+     * <br>This method does not calculate the owner in.
+     * <b>Allowed permissions override denied permissions of {@link net.dv8tion.jda.core.entities.PermissionOverride PermissionOverrides}!</b>
+     *
+     * <p>All permissions returned are explicitly granted to this Member via its {@link net.dv8tion.jda.core.entities.Role Roles}.
+     * <br>Permissions like {@link net.dv8tion.jda.core.Permission#ADMINISTRATOR Permission.ADMINISTRATOR} do not
+     * grant other permissions in this value.
+     * <p>This factor in all {@link net.dv8tion.jda.core.entities.PermissionOverride PermissionOverrides} that affect this member
+     * and only grants the ones that are explicitly given.
+     *
+     * @param  channel
+     *         The target channel of which to check {@link net.dv8tion.jda.core.entities.PermissionOverride PermissionOverrides}
+     * @param  member
+     *         The non-null {@link net.dv8tion.jda.core.entities.Member Member} for which to get implicit permissions
+     *
+     * @throws IllegalArgumentException
+     *         If any of the arguments is {@code null}
+     *         or the specified entities are not from the same {@link net.dv8tion.jda.core.entities.Guild Guild}
+     *
+     * @return Primitive (unsigned) long value with the implicit permissions of the specified member in the specified channel
+     *
+     * @since  3.1
+     */
+    public static long getImplicitPermission(Channel channel, Member member)
```

Comment:
The reason I did not do this is because in `getEffectivePermission` it is able to return without knowing all implicit permissions.
e.g. it can return if the public role has `Permission.ADMINISTRATOR` but `getImplicitPermission` retrieves **all** permissions and then returns which would make `getEffectivePermission` less efficient.